sequencing_report_service/nextflow.py

```diff
@@ -2,106 +2,174 @@
+
+        Returns
+        -------
+        config: dict
+       """
         try:
-            self._cmd = ['nextflow', '-config',
-                         config_dict['nf_config'],
-                         'run',
-                         config_dict['main_workflow_path'],
-                         '-profile',
-                         config_dict['nf_profile']]
-            if not config_dict.get('parameters'):
-                raise NextflowConfigError("The parameters to the nextflow job was empty.")
-            self._raw_params = config_dict['parameters']
-            self._config_dict = config_dict
-        except KeyError as exc:
-            log.error(exc)
-            raise NextflowConfigError(exc) from exc
-        except NextflowConfigError as exc:
-            log.error(exc)
-            raise exc
-
-    def _construct_nf_param_list(self, runfolder_path):
-        """
-        Read the parameters part of the dict and use it to generate a list
-        of parameters and their values.
-        :param: runfolder_path path to the runfolder to process
-        """
+            with open(self.config_dir / f"{pipeline}.yml", "r") as config_file:
+                config = yaml.safe_load(config_file.read())
+            with open(self.config_dir / "schema.json", "r") as pipeline_config_schema:
+                schema = json.load(pipeline_config_schema)
+            jsonschema.validate(config, schema)
+        except (FileNotFoundError, jsonschema.ValidationError):
+            log.exception('')
+            raise
 
-        # This uses the variable interpolation capablities of the ConfigParser
-        # to replace variables setup in the config. Creating a new dict is necessary
-        # to conform to the standard format required by the Python config parser.
-        # /JD 2018-04-03
-        defaults = {'runfolder_path': str(runfolder_path),
-                    'runfolder_name': runfolder_path.name,
-                    'current_year': datetime.datetime.now().year}
-        conf = configparser.ConfigParser(
-                defaults=defaults,
-                interpolation=configparser.ExtendedInterpolation())
-        params_as_conf_dict = {'nextflow_config': self._raw_params}
-        conf.read_dict(params_as_conf_dict)
-
-        lst = []
-        for key, value in conf['nextflow_config'].items():
-            # This check is necessary to ensure that the default values
-            # are not also output here. /JD 2019-04-03
-            if key in self._raw_params.keys():
-                lst += [f"--{key}", f"{value}"]
-
-        return lst
-
-    def _construct_environment(self, runfolder_path):
+        runfolder_path = Path(runfolder_path)
+        defaults = {
+            "current_year": datetime.datetime.now().year,
+            "runfolder_path": str(runfolder_path),
+            "runfolder_name": runfolder_path.name,
+        }
+
+        input_samplesheet = input_samplesheet or config.get("input_samplesheet", "")
```

Comment:
Yeah I agree, although I don't really have a better solution. Are the samplesheets for the analysis pipelines much bigger than the runfolder samplesheets? I think there is the same issue in arteria-bcl2fastq and the same solution has been implemented there (see https://github.com/arteria-project/arteria-bcl2fastq/blob/afb1332c016d7af99cb710d3c6f4fe8f10775422/bcl2fastq/handlers/bcl2fastq_handlers.py#L165)

Reply:
yeah, I couldn't really suggest a better approach either, unless implementing a lot of extra functionality to handle this. In theory (and sometimes in practice), the analysis sample sheets can. be much bigger since they can combine samples from multiple runfolders. But I think that's only a problem in theory, in reality I don't think it will be a problem.